Mage.Sets/src/mage/cards/u/UnfinishedBusiness.java

```diff
@@ -0,0 +1,123 @@
+        super(card);
+    }
+
+    @Override
+    public UnfinishedBusiness copy() {
+        return new UnfinishedBusiness(this);
+    }
+}
+
+class UnfinishedBusinessEffect extends OneShotEffect{
+
+    public UnfinishedBusinessEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "Return target creature card from your graveyard to the battlefield, then return up to two target Aura and/or Equipment cards from your graveyard to the battlefield attached to that creature. (If the Auras can not enchant that creature, they remain in your graveyard.)";
+    }
+
+    private UnfinishedBusinessEffect(final UnfinishedBusinessEffect effect) {super(effect);}
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer((source.getControllerId()));
+        if (controller == null){
+            return false;
+        }
+
+        // Return target creature from the graveyard to the battlefield
+        Card targetCreature = game.getCard(source.getTargets().getFirstTarget());
+        if (targetCreature != null){
+            controller.moveCards(targetCreature, Zone.BATTLEFIELD, source, game);
+        }
+        Permanent permanentCreature = targetCreature == null ? null : game.getPermanent(source.getTargets().getFirstTarget());
+
+        // Target auras and/or equipment in your graveyard.
+        Cards cardsInitial = new CardsImpl(source.getTargets().get(1).getTargets());
+        if (cardsInitial.isEmpty()) {
+            return false;
+        }
+
+        // Auras that cannot be attached to the creature stay in the graveyard
+        // Create a list of legal cards to return
+        Cards cards = new CardsImpl();
+        for(UUID c: cardsInitial){
+            if (game.getCard(c).hasSubtype(SubType.EQUIPMENT,game)){
+                // always add equipment cards
+                cards.add(c);
+            }
+            else if (permanentCreature != null &&
+                    !permanentCreature.cantBeAttachedBy(game.getCard(c),source, game, false) &&
+                    game.getCard(c).hasSubtype(SubType.AURA, game)){
+                // only add auras if the creature has returned
+                // add auras that can be attached to creature
+                cards.add(c);
+            }
+        }
+        if (cards.isEmpty()){
+            return false;
+        }
+
+        // Handle return of legal auras and equipment
+        if (permanentCreature != null){
+            cards.getCards(game)
+                    .forEach(card -> game.getState().setValue("attachTo:" + card.getId(), permanentCreature));
+        }
+        controller.moveCards(cards, Zone.BATTLEFIELD, source, game);
+        if (permanentCreature != null){
+            for(UUID id: cards){
+                permanentCreature.addAttachment(id, source, game);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public Effect copy() {
```

Comment:
Return type of the method should be `UnfinishedBusinessEffect` rather than just `Effect`.